core/unix/os.c

```diff
@@ -8280,8 +8289,8 @@ post_system_call(dcontext_t *dcontext)
 #endif
 #ifdef LINUX
     case SYS_ppoll: {
-        handle_post_extended_syscall_sigmasks(dcontext, success);
-        set_syscall_param(dcontext, 3, dcontext->sys_param3);
+        if (handle_post_extended_syscall_sigmasks(dcontext, success))
+            set_syscall_param(dcontext, 3, dcontext->sys_param3);
```

Comment:
Using sys_param3 is owned by os.c, so it shouldn't rely on a signal.c function to tell whether to use it, esp when we never called handle_pre_extended_syscall_sigmask: seems like a modularity breach.  Seems cleaner to set sys_param3 in pre always and check for NULL here, right?

Reply:
Ok! Except for pselect6, I am adding a small trick, since the syscall is getting a pointer to memory where the sigmask is stored, I am abusing the parameter that actually holds this pointer and setting this to NULL.